lib/in_out/reader.py

```diff
@@ -30,48 +51,40 @@ def linux_input(log_directory, channel_list, starting_date, ending_date):
     logs = defaultdict(list)
     
     #enter year folder
-    for year_iterator in range(starting_year, ending_year + 1):  
+    for year_iterator in range(starting_year, ending_year + 1):
         #enter month folders
         for month_iterator in range(starting_month if year_iterator == starting_year else 1, ending_month + 1 if year_iterator == ending_year else 13):
-            extra0 = "0" if month_iterator < 10 else ""        
+            extra0 = "0" if month_iterator < 10 else ""
             #enter day folder
             for day_iterator in range(starting_day if (month_iterator == starting_month and year_iterator == starting_year) else 1, ending_day + 1 if (month_iterator == ending_month and year_iterator==ending_year) else 32):
                 extra0_2 = "0" if day_iterator < 10 else ""
                 work_path = log_directory+str(year_iterator)+"/"+extra0 + \
                     str(month_iterator) + "/" + extra0_2 + \
                     str(day_iterator) + "/"
                 #select relevant channels
-                if not os.path.exists(work_path):
-                    if not((month_iterator == 2 and (day_iterator == 29 or day_iterator == 30 or day_iterator == 31)) or ((month_iterator == 4 or month_iterator == 6 or month_iterator == 9 or month_iterator == 11) and day_iterator == 31)):
-                        print "[Error | io/linuxInput] Path", work_path, "doesn't exist"
-                else:
-                    for channel_searched in os.listdir(work_path):
-                        channel_name = channel_searched[:-4]
-                        if channel_name in channel_list or (len(channel_list) == 1 and channel_list[0] == "ALL"):
-                            file_path = work_path + channel_name + ".txt"
-                            if not os.path.exists(file_path):
-                                print "[Error | io/linuxInput] Channel " + file_path + " doesn't exist"
-                                continue
-                            with open(file_path) as f:
-                                """ day_data stores all the lines of the file channel_name """
-                                if config.DEBUGGER:
-                                    print "Working on: " + file_path
-                                
-                                day_data = f.readlines()             
-                            
-                            f.close()
+                channel_list = check_if_path_and_file_is_valid(work_path, channel_list)
+
+                for channel_name in channel_list:
+                    file_path = work_path + channel_name + ".txt"
+                    with open(file_path) as f:
+                        """ day_data stores all the lines of the file channel_name """
+                        if config.DEBUGGER:
+                            m = re.match('(?P<machine_specific>.*/)(?P<universal>IRCLogParser/.*)', file_path)
```

Comment:
@Kausam Please add a comment specifying two example patterns this regex would match.

Reply:
As a rule regex should be as specific as possible in order to avoid potential mismatches. If you can give a more precise pattern here, you should definitely do that.